test/storage/slab/check_slab.c

```diff
@@ -116,6 +113,96 @@ START_TEST(test_insert_large)
 END_TEST
 
 /**
+ * Tests item_reserve, item_backfill and item_release
+ */
+START_TEST(test_reserve_backfill_release)
+{
+#define KEY "key"
+    struct bstring key, val;
+    item_rstatus_t status;
+    struct item *it;
+    uint32_t vlen, dataflag = 12345;
+
+    test_reset();
+
+    key = str2bstr(KEY);
+
+    vlen = 1000 * KiB;
+    val.len = vlen / 2;
+    val.data = cc_alloc(val.len);
+    cc_memset(val.data, 'A', val.len);
+
+    /* reserve */
+    time_update();
+    status = item_reserve(&it, &key, &val, vlen, dataflag, 0);
+    free(val.data);
+    ck_assert_msg(status == ITEM_OK, "item_reserve not OK - return status %d", status);
+
+    ck_assert_msg(it != NULL, "item_reserve returned NULL object");
+    ck_assert_msg(!it->is_linked, "item linked by mistake");
+    ck_assert_msg(!it->in_freeq, "linked item with key %.*s in freeq", key.len, key.data);
+    ck_assert_msg(!it->is_raligned, "item with key %.*s is raligned", key.len, key.data);
+    ck_assert_int_eq(it->klen, sizeof(KEY) - 1);
+    ck_assert_int_eq(it->dataflag, dataflag);
+    ck_assert_int_eq(it->vlen, val.len);
+    ck_assert_msg(strspn(item_data(it), "A") == val.len, "item_data contains wrong value %.*s", it->vlen, item_data(it));
+
+    /* backfill */
+    val.len = vlen - val.len;
+    val.data = cc_alloc(val.len);
+    cc_memset(val.data, 'B', val.len);
+    item_backfill(it, &val, false);
+    free(val.data);
+    ck_assert_msg(!it->is_linked, "item linked by mistake");
+    ck_assert_int_eq(it->vlen, vlen);
+    ck_assert_msg(strspn(item_data(it) + vlen - val.len, "B") == val.len,
```

Comment:
add an assertion on the first part of the item_data to make sure it was not modified?